Feature/mobile navigation header

weberjavi opened the pull request:
This PR creates a mobile navigation menu as proposed by design team. Designs can be checked [here](https://app.zeplin.io/project/598b1dc89689e51e1cd1b5ce/screen/5a5dc4f2556f113ea85894cc) 
Some changes from the original design have been made (after agreement with design team):  
* Nav elements that contain nested routes do not display the arrow icon (are always opened) . 
* An underline is displayed on the actual section to have a navigation reference when the menu is open.  
* Scroll functionality in the home page (that enables changing in opacity to the mobile navbar) is not developed in this stage.  
  
![navbarmobile](https://user-images.githubusercontent.com/6906348/35566346-5bd3c730-05c1-11e8-892e-1a751f8027d5.gif)

weberjavi commented on app/javascript/app/components/navbar-mobile/nav-with-child-menu/nav-with-child-menu.js:
```diff
@@ -0,0 +1,7 @@
+import { connect } from 'react-redux';
+
+import { actions } from 'components/hamburger';
```
:-)

weberjavi commented on app/javascript/app/components/hamburger/hamburger-component.jsx:
```diff
@@ -0,0 +1,45 @@
+  <div className={styles.container}>
+    <p className={cx(styles.text, className)}>{text}</p>
+    {isOpen && (
+      <div
```
I agree

Bluesmile82 commented:
Please take a look at the anchor fixed bar when we scroll down. Not sure if it should go down or just disappear in mobile. We can talk with design about it.
![image](https://user-images.githubusercontent.com/9701591/35581098-1ed3e7aa-05eb-11e8-8f91-1eae5992bec3.png)
![image](https://user-images.githubusercontent.com/9701591/35581138-37330a7e-05eb-11e8-9871-84a6c7d40fa5.png)

j8seangel commented:
There are still some differences with the designs, for example, the nav with child menu still needs styles to match with design, uppercase text, letter-spacing...

After that feel free to merge!

weberjavi commented:
Addressed design-implementation inconsistencies (letter spacing and casing).  
Also fixed a couple of bugs detected by @Bluesmile82 👍   (sticky nav anchors positioning and share button reverse behaviour)
